client/lib/embed-frame-markup/index.jsx

```diff
@@ -46,25 +43,19 @@ export default function generateEmbedFrameMarkup( { body, scripts, styles } = {}
 				`,
 					} }
 				/>
-				{ createFragment(
-					mapValues( scripts, script => {
-						let extra;
-						if ( script.extra ) {
-							extra = (
-								<script
-									dangerouslySetInnerHTML={ {
-										__html: script.extra,
-									} }
-								/>
-							);
-						}
-
-						return createFragment( {
-							extra: extra,
-							script: <script src={ script.src } />,
-						} );
-					} )
-				) }
+				{ map( scripts, ( { extra, src }, key ) => {
+					return [
```

Comment:
In this case we're mapping to create a nested array, e.g. `[ [ <script /> ] ]` . I think effectively there is no difference, specifically for how we're using it for rendering to string here, but in normal React renders I've observed to have abysmal performance characteristics (e.g. https://github.com/WordPress/gutenberg/pull/3008#issuecomment-336487380).

Reply:
Hmm, I guess I could use [`flatMap`](https://lodash.com/docs/4.17.4#flatMap) instead...

Reply:
30f9189